### PR TITLE
Adds a timestamp field to mutations.

### PR DESCRIPTION
**Pull request by aboodman**
Note: Format change. Not changing REPLICACHE_FORMAT_VERSION constant
because we should only do so once per release by policy, so that
should be something we do during release.

### CODE REVIEW

File: src/db/visitor.test.ts [hunk 1]
```diff
@@ -83,6 +83,7 @@ test('test that we get to the data nodes', async () => {
       fakeHash('none'),
       prevCommit.valueHash,
       prevCommit.indexes,
+      88,
```
Comment (arv) [inline src/db/visitor.test.ts:86]: Please do not use this number for tests... it is associated with a symbol of hate.

File: src/replicache.ts [hunk 2]
```diff
@@ -225,6 +227,8 @@ export class Replicache<MD extends MutatorDefs = {}> {
    */
   pusher: Pusher;
 
+  now: Now = () => performance.now();
```
Comment (arv) [inline src/replicache.ts:230]: New public API? Why is this exposed?

File: src/replicache.ts [hunk 3]
```diff
@@ -1156,6 +1163,7 @@ export class Replicache<MD extends MutatorDefs = {}> {
           name,
           mutatorImpl,
           args,
+          performance.now(),
```
Comment (arv) [inline src/replicache.ts:1166]: should this be `this.now()`?

**Review by arv — COMMENTED** [review]
Needs more work

File: src/db/visitor.test.ts (hunk 1 above)
Comment (aboodman) [reply src/db/visitor.test.ts:86]: Whoops, forgot about that. For me it always means the number of miles per hour the delorean has to go to activate the flux capacitor. Will avoid in the future.

File: src/replicache.ts (hunk 2 above)
Comment (aboodman) [reply src/replicache.ts:230]: Mistake.

File: src/replicache.ts (hunk 3 above)
Comment (aboodman) [reply src/replicache.ts:1166]: No, I thought at first that `Now` would have to be injected for testing, but you are already controlling the timers in the tests with monkey patching.

**Comment by aboodman** [comment]
Addressed in https://github.com/rocicorp/replicache/pull/783.

**Pull request merged**